src/main/java/org/broadinstitute/hellbender/tools/walkers/annotator/VariantAnnotator.java

```diff
@@ -0,0 +1,293 @@
+        // get the list of all sample names from the variant VCF input rod, if applicable
+        final  List<String> samples = getHeaderForVariants().getGenotypeSamples();
+        variantSamples = new IndexedSampleList(samples);
+
+        if ( USE_ALL_ANNOTATIONS ) {
+            annotatorEngine = VariantAnnotatorEngine.ofAllMinusExcluded(variantAnnotationArgumentCollection.annotationsToExclude, dbsnp.dbsnp, comps, false);
+        } else {
+            annotatorEngine = VariantAnnotatorEngine.ofSelectedMinusExcluded(variantAnnotationArgumentCollection, dbsnp.dbsnp, comps, false);
+        }
+        annotatorEngine.addExpressions(expressionsToUse, resources, expressionAlleleConcordance );
+
+        // setup the header fields
+        // note that if any of the definitions conflict with our new ones, then we want to overwrite the old ones
+        final Set<VCFHeaderLine> hInfo = new HashSet<>();
+
+        hInfo.addAll(annotatorEngine.getVCFAnnotationDescriptions(false));
+        hInfo.addAll(getHeaderForVariants().getMetaDataInInputOrder());
+
+        // for the expressions, pull the info header line from the header of the resource rod
+        for ( final VariantAnnotatorEngine.VAExpression expression : annotatorEngine.getRequestedExpressions() ) {
+            // special case the ID field
+            if ( expression.fieldName.equals("ID") ) {
+                hInfo.add(new VCFInfoHeaderLine(expression.fullName, 1, VCFHeaderLineType.String, "ID field transferred from external VCF resource"));
+            } else {
+                final VCFInfoHeaderLine targetHeaderLine = ((VCFHeader) getHeaderForFeatures(expression.binding)).getInfoHeaderLines().stream()
+                            .filter(l -> l.getID().equals(expression.fieldName))
+                            .findFirst().orElse(null);
+
+                VCFInfoHeaderLine lineToAdd;
+                if (targetHeaderLine != null) {
+                    expression.hInfo = targetHeaderLine;
+                    if (targetHeaderLine.getCountType() == VCFHeaderLineCount.INTEGER) {
+                        lineToAdd = new VCFInfoHeaderLine(expression.fullName, targetHeaderLine.getCount(), targetHeaderLine.getType(), targetHeaderLine.getDescription());
+                    } else {
+                        lineToAdd = new VCFInfoHeaderLine(expression.fullName, targetHeaderLine.getCountType(), targetHeaderLine.getType(), targetHeaderLine.getDescription());
+                    }
+                } else {
+                    lineToAdd = new VCFInfoHeaderLine(expression.fullName, VCFHeaderLineCount.UNBOUNDED, VCFHeaderLineType.String, "Value transferred from another external VCF resource");
+                }
+                hInfo.add(lineToAdd);
+                expression.hInfo = lineToAdd;
+            }
+        }
+
+        // TODO ask reviewer, VCFUtils.withUpdatedContigs is what GATK3 calls into, it isn't used anywhere in 4 though so should it be used here?
```

Comment:
Git blame reveals that the updated contigs call goes way back.  It may just be a cleanup to protect against bespoke, handcrafted, artisanal VCFs that don't have the right sequence dictionary.  Hopefully in GATK4 we do a more rigorous check up front rather than just clean up?  withUpdatedContigs can go horribly wrong if someone accidentally supplies an hg19 reference with a hg38 VCF and now that we have a lot of data on each it's very likely to happen.